ai-catalog/samples/gemini-live-todo/src/main/java/com/android/ai/samples/geminilivetodo/ui/TodoScreen.kt

```diff
@@ -0,0 +1,225 @@
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.android.ai.samples.geminilivetodo.R
+import com.android.ai.samples.geminilivetodo.data.Todo
+import kotlin.collections.reversed
+
+/**
+ * The main screen for the To-do list application.
+ * This composable is stateful, connecting to the ViewModel to manage UI state and events.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@RequiresPermission(Manifest.permission.RECORD_AUDIO)
+@Composable
+fun TodoScreen(viewModel: TodoScreenViewModel = hiltViewModel()) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    var text by remember { mutableStateOf("") }
+
+    LaunchedEffect(Unit) {
+        viewModel.initializeGeminiLive()
+    }
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                colors = TopAppBarDefaults.topAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.primaryContainer,
+                    titleContentColor = MaterialTheme.colorScheme.primary,
+                ),
+                title = { Text(stringResource(R.string.gemini_live_title)) },
+            )
+        },
+        floatingActionButton = {
+            if (uiState is TodoScreenUiState.Success) {
+                val successState = uiState as TodoScreenUiState.Success
+                val micIcon = when {
+                    !successState.isLiveSessionReady -> Icons.Filled.MicOff
+                    successState.isLiveSessionRunning -> Icons.Filled.Mic
+                    else -> Icons.Filled.MicNone
+                }
+
+                val containerColor = if (successState.isLiveSessionRunning) {
+                    val infiniteTransition =
+                        rememberInfiniteTransition(label = "mic_color_transition")
+                    infiniteTransition.animateColor(
+                        initialValue = MaterialTheme.colorScheme.primary.copy(alpha = 0.2f),
+                        targetValue = MaterialTheme.colorScheme.primary.copy(alpha = 0.9f),
+                        animationSpec = infiniteRepeatable(
+                            animation = tween(1000, easing = LinearEasing),
```

Comment:
Did Ivy tell you to use Linear? If not I would just use the default, it's generally nicer than Linear